cmd/state-update-dialog/frontend/main.js

```diff
@@ -1,17 +1,45 @@
 // Main entry point
 function start() {
-    showContextMenu = document.oncontextmenu;
+    let showContextMenu = document.oncontextmenu;
     document.oncontextmenu = () => false;
     backend.Bindings.DebugMode().then(debugMode => {
         console.log(debugMode);
         if (debugMode) document.oncontextmenu = showContextMenu; // Enable context menu if we're running in debug mode
     })
 
     Promise.all([
-        window.backend.Bindings.CurrentVersion(),
-        window.backend.Bindings.AvailableVersion()
+        backend.Bindings.CurrentVersion(),
+        backend.Bindings.AvailableVersion()
     ]).then(result => {
-        document.body.append(document.createTextNode("Current: " + result[0] + ", Available: " + result[1]));
+        document.getElementById("CurrentVersion").innerText = result[0];
+        document.getElementById("AvailableVersion").innerText = result[1];
+    })
+
+    backend.Bindings.Warning().then(result => {
+        if (result === "") return;
+        document.getElementById("warning-wrapper").style.display = "block";
+        document.getElementById("warning").innerHTML = result;
+    });
+
+    populateChangelog();
+
+    document.getElementById("close-btn").addEventListener("click", () => backend.Bindings.Exit());
+}
+
+function populateChangelog(tries) {
+    tries = tries || 0;
+    if (tries > 10) {
+        return;
```

Comment:
I'm probably misunderstanding, but, at a glance, I think this should be `if (tries > 10) { tries = 10 }`

Reply:
No, I don't want to try more than 10 times.